pandas/plotting/_matplotlib/tools.py

```diff
@@ -23,13 +23,24 @@
     from matplotlib.table import Table
 
 
+def do_adjust_figure(fig):
+    if not hasattr(fig, "get_constrained_layout"):
+        return False
+    return not fig.get_constrained_layout()
+
+
+def maybe_adjust_figure(fig, *args, **kwargs):
+    if do_adjust_figure(fig):
```

Comment:
can you add a doc-string (explaining the purpose here) & type what you can (e.g. fig)

Reply:
Sure.